specification/storage/resource-manager/Microsoft.Storage/stable/2022-09-01/networkSecurityPerimeter.json

```diff
@@ -0,0 +1,416 @@
+      "readOnly": true,
+      "properties": {
+        "name": {
+          "description": "Name of the resource",
+          "type": "string"
+        },
+        "properties": {
+          "description": "Properties of Access Rule",
+          "readOnly": true,
+          "type": "object",
+          "properties": {
+            "direction": {
+              "description": "Direction of Access Rule",
+              "enum": [
+                "Inbound",
+                "Outbound"
+              ],
+              "type": "string",
+              "x-ms-enum": {
+                "name": "NspAccessRuleDirection",
+                "modelAsString": true
+              }
+            },
+            "addressPrefixes": {
+              "description": "Address prefixes in the CIDR format for inbound rules",
+              "type": "array",
+              "items": {
+                "description": "IP Address",
+                "type": "string"
+              }
+            },
+            "subscriptions": {
+              "description": "Subscriptions for inbound rules",
+              "type": "array",
+              "items": {
+                "description": "Subscription for inbound rule",
+                "type": "object",
+                "readOnly": true,
+                "properties": {
+                  "id": {
+                    "description": "The ARM identifier of subscription",
+                    "type": "string"
+                  }
+                }
+              }
+            },
+            "networkSecurityPerimeters": {
+              "description": "NetworkSecurityPerimeters for inbound rules",
+              "type": "array",
+              "readOnly": true,
+              "items": {
+                "$ref": "#/definitions/NetworkSecurityPerimeter"
+              }
+            },
+            "fullyQualifiedDomainNames": {
+              "description": "FQDN for outbound rules",
+              "type": "array",
+              "readOnly": true,
+              "items": {
+                "description": "Fully qualified domain name",
+                "type": "string"
+              }
+            }
+          }
+        }
+      }
+    },
+    "NetworkSecurityPerimeter": {
+      "description": "NetworkSecurityPerimeter related information",
+      "type": "object",
+      "readOnly": true,
+      "properties": {
+        "id": {
+          "description": "The ARM identifier of the resource",
+          "type": "string"
+        },
+        "perimeterGuid": {
+          "description": "Guid of the resource",
+          "type": "string"
+        },
+        "location": {
+          "description": "Location of the resource",
+          "type": "string"
+        }
+      }
+    }
+  },
+  "parameters": {
+    "resourceAssociationId": {
+      "name": "resourceAssociationId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The unique Network Security Perimeter association id",
```

Comment:
(Optional) It would be better to give more context for how to prepare this parameter, and the is there any limitation of Network Security Perimeter used in this parameter?

Reply:
What you suggested make sense in the normal case. While Network security Perimeter is a complex concept, the association id is not generated by storage either, user need to create a network security perimeter association, then they will have an association id, then the association id can be used in storage NSP Apis. Most stories happen in Network team side, customer will understand these concepts from there first, before they use any of these Apis. It will be very hard(also not necessary) for storage to explain the whole flow here in the swagger.

Reply:
@Elvis-Shi 
The description will be in storage rest API doc/ SDK, and PSH/CLI parameter descriptions.
So are you OK with the description in all PSH/CLI/SDK and rest API doc?

From my point of view, it would be better if we can give some hint here how to prepare the parameter.
However, if we all customer try to use this API should already know the concept of NSP, I am good with it.